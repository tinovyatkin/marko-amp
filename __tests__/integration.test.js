'use strict';

const amphtmlValidator = require('amphtml-validator');
const Template = require('marko').load(
  require.resolve('./templates/amp-full-render-doc.marko'),
);

describe('Full page render and validation', () => {
  it('render and validate AMP version', async () => {
    const render = await Template.render({
      $global: {
        amp: true,
      },
    }).toString();

    expect(render).toMatchSnapshot();

    // validate
    const validator = await amphtmlValidator.getInstance();
    const result = await validator.validateString(render);
    expect(result.errors).toHaveLength(0);
    expect(result).toHaveProperty('status', 'PASS');
  });
});
