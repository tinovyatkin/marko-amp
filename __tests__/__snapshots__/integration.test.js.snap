// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full page render and validation render and validate AMP version 1`] = `
"<!DOCTYPE html><html âš¡><head><style amp-boilerplate>
    body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}
  </style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript><script async src=\\"https://cdn.ampproject.org/v0.js\\"></script><script async custom-element=\\"amp-video\\" src=\\"https://cdn.ampproject.org/v0/amp-video-0.1.js\\"></script><link rel=\\"canonical\\" href=\\"full-page-render.html\\"><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width\\"></head><body><amp-img src=\\"images/mountains.webp\\" alt=\\"Mountains\\" height=\\"368\\" width=\\"550\\"><amp-img src=\\"images/mountains.jpg\\" alt=\\"Mountains\\" height=\\"368\\" width=\\"550\\" fallback></amp-img></amp-img><amp-video poster=\\"images/kitten-playing.png\\" controls height=\\"360\\" width=\\"640\\" layout=\\"responsive\\"><source src=\\"videos/kitten-playing.webm\\" type=\\"video/webm\\"><source src=\\"videos/kitten-playing.mp4\\" type=\\"video/mp4\\"><div fallback><p>This browser does not support the video element.</p></div></amp-video></body></html>"
`;
